android/app/src/main/java/org/fossasia/openevent/viewmodels/auth/ChangePasswordActivityViewModel.java

```diff
@@ -0,0 +1,101 @@
+package org.fossasia.openevent.viewmodels.auth;
+
+import android.arch.lifecycle.LiveData;
+import android.arch.lifecycle.MutableLiveData;
+import android.arch.lifecycle.ViewModel;
+
+import org.fossasia.openevent.api.APIClient;
+import org.fossasia.openevent.data.auth.Login;
+import org.fossasia.openevent.data.auth.User;
+import org.fossasia.openevent.dbutils.RealmDataRepository;
+import org.fossasia.openevent.utils.AuthUtil;
+import org.fossasia.openevent.utils.JWTUtils;
+import org.json.JSONException;
+
+import io.reactivex.android.schedulers.AndroidSchedulers;
+import io.reactivex.disposables.CompositeDisposable;
+import io.reactivex.schedulers.Schedulers;
+import timber.log.Timber;
+
+public class ChangePasswordActivityViewModel extends ViewModel {
+
+    public static final int ON_ERROR = 2;
+    public static final int ON_COMPLETE = 3;
+
+    private RealmDataRepository realmRepo;
+    private User user;
+    private LiveData<User> userLiveData;
+    private MutableLiveData<Integer> checkPasswordResponse;
+    private MutableLiveData<Integer> changePasswordResponse;
+    private final CompositeDisposable compositeDisposable;
+
+    public ChangePasswordActivityViewModel() {
+        realmRepo = RealmDataRepository.getDefaultInstance();
+        compositeDisposable = new CompositeDisposable();
+    }
+
+    public LiveData<User> getUser() {
+        if (userLiveData == null) {
+            user = realmRepo.getUser();
+            user.addChangeListener(realmModel -> {});
```

Comment:
Why this additional change litener?

Reply:
I checked the method ```getUser```. The comment above it states to add a change listener.
``` Java
/**
     * Returns Future style User which is null
     * To get the contents of User, add an OnRealmChangeListener
     * which notifies about the object state asynchronously
     *
     * @return User Returns User Future
*/
```

But when i removed the changeListener, still i am able to get it work. If user is null, then how does it still work ?

Reply:
It isn't null. If it'd be null, you would have gotten NPE. It's an error in comment. I didn't check the implementation of the method, so if it is different than the rest, i.e., not asynchronous, maybe we should change the method.

Also, adding change listener is what is done automatically by LiveData wrapper

Reply:
It is asynchronous.
 > adding change listener is what is done automatically by LiveData wrapper

Yeah, i missed this point.